code/game/objects/structures/dropship_equipment.dm

```diff
@@ -877,42 +927,356 @@
+	user.pixel_y = 8
+	var/matrix/reset_matrix = user.transform
+	animate(user, 3, transform = reset_matrix, pixel_y = 0, alpha = pre_rappel_alpha, flags = ANIMATION_PARALLEL)
+	user.canmove = TRUE
+
+///Deploys the rappel and unlocks the hatch so that people can drop down
+/obj/structure/dropship_equipment/rappel_system/proc/deploy_rope(turf/target)
+	if(!rope || rappel_condition < RAPPEL_CONDITION_GOOD || rappel_state >= RAPPEL_STATE_USABLE)
+		return
+
+	rope.forceMove(target)
+	rappel_state = RAPPEL_STATE_USABLE
+	update_icon_state()
+	rope.update_icon_state()
+	flick("rope_deploy", rope)
+
+	playsound(target, 'sound/effects/tadpolehovering.ogg', 100, TRUE, falloff = 2.5)
+	playsound(target, 'sound/effects/rappel.ogg', 50, TRUE)
+	playsound(src, 'sound/effects/rappel.ogg', 50, TRUE)
+	target.balloon_alert_to_viewers("You see a dropship fly overhead and begin dropping ropes!")
+	balloon_alert_to_viewers("You hear a hiss as [src] unlocks!")
+
+
+///Feedback for when PO manually retracts the rope. Leads back into retract_rope after sounds and balloon alerts are done.
+/obj/structure/dropship_equipment/rappel_system/proc/pre_retract()
+	if(rappel_state == RAPPEL_STATE_RETRACTING) //We're already retracting!
+		return
+
+	rappel_state = RAPPEL_STATE_RETRACTING
+	update_icon_state()
+	rope.update_icon_state()
+
+	playsound(src, 'sound/machines/hiss.ogg', 25)
+	balloon_alert_to_viewers("The system hums as ropes start reeling in.")
+	rope.balloon_alert_to_viewers("The rope starts reeling into the sky...")
+
+	addtimer(CALLBACK(src, .proc/retract_rope), 5 SECONDS)
+
+
+///Undeploys the rappel and locks the hatch. Rappel cannot be retracted if it is currently being attacked (hooked)
+/obj/structure/dropship_equipment/rappel_system/proc/retract_rope()
+	if(rappel_state != RAPPEL_STATE_RETRACTING)
+		return
+
+	if(rappel_condition == RAPPEL_CONDITION_HOOKED)
+		rappel_state = RAPPEL_STATE_USABLE
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 25)
+		return
+
+	rappel_state = RAPPEL_STATE_LOCKED
+	update_icon_state()
+	var/turf/target = get_turf(rope)
+	target.balloon_alert_to_viewers("A dropship overhead retracts the ropes!")
+	balloon_alert_to_viewers("The rappel ropes reel back into [src], locking the system with a click!")
+	playsound(target, 'sound/effects/tadpolehovering.ogg', 100, TRUE, falloff = 2.5)
+	playsound(target, 'sound/effects/rappel.ogg', 50, TRUE)
+	playsound(src, 'sound/effects/rappel.ogg', 50, TRUE)
+	flick("rope_up", rope)
+
+	addtimer(CALLBACK(src, .proc/reel_in), 0.4 SECONDS)
+
+///Part 2 of retract_rope(), moves the rope back into the system after the rope animation has completed. Unbuckles any mobs which were attached to it.
+/obj/structure/dropship_equipment/rappel_system/proc/reel_in()
+	rope.forceMove(src)
+
+///Handles xeno attacks on the system; called by the rappel rope whenever attack_alien() is called on the rope
+/obj/structure/dropship_equipment/rappel_system/proc/buckle(mob/living/carbon/xenomorph/X)
```

Comment:
don't use X.
Also this probably should have another name since its just confusing it with actual buckling.